provider/providerutil/providerutil.go

```diff
@@ -151,3 +160,115 @@ func GetXCUITestDriverVersion() (string, error) {
 func GetUiAutomator2DriverVersion() (string, error) {
 	return GetAppiumDriverVersion("uiautomator2")
 }
+
+// Check if sdb is available on the host by checking its version
+func SdbAvailable() bool {
+	logger.ProviderLogger.LogInfo("provider_setup", "Checking if sdb is set up and available on the host PATH")
+
+	cmd := exec.Command("sdb", "version")
+	err := cmd.Run()
+	if err != nil {
+		logger.ProviderLogger.LogDebug("provider_setup", fmt.Sprintf("sdbAvailable: sdb is not available or command failed - %s", err))
+		return false
+	}
+	return true
+}
+
+// Check if chromedriver is located in the drivers provider folder
+func isChromeDriverAvailable() bool {
+	var chromedriverPath string
+	if runtime.GOOS == "windows" {
+		chromedriverPath = fmt.Sprintf("%s/drivers/chromedriver.exe", config.ProviderConfig.ProviderFolder)
+	} else {
+		chromedriverPath = fmt.Sprintf("%s/drivers/chromedriver", config.ProviderConfig.ProviderFolder)
+	}
+
+	_, err := os.Stat(chromedriverPath)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return err == nil
+}
+
+func CheckChromeDriverAndDownload() error {
+	if isChromeDriverAvailable() {
+		logger.ProviderLogger.LogInfo("provider_setup", "Chromedriver is available in the drivers provider folder, it will not be downloaded.")
+		return nil
+	}
+
+	var url string
+	switch runtime.GOOS {
+	case "linux":
+		url = "https://chromedriver.storage.googleapis.com/2.36/chromedriver_linux64.zip"
+	case "darwin":
+		url = "https://chromedriver.storage.googleapis.com/2.36/chromedriver_mac64.zip"
+	case "windows":
+		url = "https://chromedriver.storage.googleapis.com/2.36/chromedriver_win32.zip"
+	default:
+		return fmt.Errorf("unsupported platform: %s", runtime.GOOS)
+	}
+
+	// Download the zip file
+	response, err := http.Get(url)
+	if err != nil {
+		return fmt.Errorf("failed to download ChromeDriver: %s", err)
+	}
+	defer response.Body.Close()
+
+	// Create a temporary file
+	tempFile, err := os.CreateTemp("", "chromedriver.zip")
+	if err != nil {
+		return fmt.Errorf("failed to create temp file: %s", err)
+	}
+	defer os.Remove(tempFile.Name())
+
+	// Write the response body to the temp file
+	_, err = io.Copy(tempFile, response.Body)
+	if err != nil {
+		return fmt.Errorf("failed to write to temp file: %s", err)
+	}
+
+	// Define the extraction directory as the provider folder
+	extractionDir := fmt.Sprintf("%s/drivers", config.ProviderConfig.ProviderFolder)
+
+	// Extract the zip file
+	err = unzip(tempFile.Name(), extractionDir) // Specify the extraction directory
+	if err != nil {
+		return fmt.Errorf("failed to extract ChromeDriver: %s", err)
+	}
+
+	return nil
+}
+
+func unzip(src string, dest string) error {
```

Comment:
Can you move this as a common function in `common\utils`

Reply:
Done in [commit](https://github.com/shamanec/GADS/pull/208/commits/bd812e76d9a05ec4df967942d1a7c8f73505efab)